zipline-bytecode/src/main/kotlin/app/cash/zipline/bytecode/types.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.bytecode
+
+import okio.Buffer
+import okio.ByteString
+
+sealed class JsObject
+
+object JsNull : JsObject()
+
+object JsUndefined : JsObject()
+
+data class JsBoolean(val value: Boolean) : JsObject()
```

Comment:
Can change these into `value` classes?

Reply:
Ooooh I hadn’t even considered value classes. That won’t work due to the sealed class.

Reply:
Ah crap, okay